[Feature] SMP-915: Integración del modulo de ubicaciones en el plugin de Cordova

Jaquilar opened the pull request.

Jaquilar committed:
SMP-915: Integración del modulo de ubicaciones en el plugin de Cordova

- Removed start Koin from auth module

- Added Location Module

Jaquilar committed:
Merge branch 'developer' into task/integrate-location-module

Jaquilar committed:
[Improvement] SMP-915: Integración del modulo de ubicaciones en el plugin de Cordova

- Implemented the Location module into the plugin

- Updated the list of actions into CordovaPlugin

brodriguezQXD commented on src/android/org/apache/cordova/plugin/sentinel/CordovaPluginSentinelSDK.kt:
```diff
@@ -0,0 +1,76 @@
+   override fun execute(
+         action: String,
+         args: JSONArray,
+         callbackContext: CallbackContext
```
What about promises, can't we handle promises here?

Jaquilar replied:
execute method just override callbacks

Done

brodriguezQXD commented on src/android/org/apache/cordova/plugin/sentinel/CordovaPluginSentinelSDK.kt:
```diff
@@ -0,0 +1,76 @@
+         args: JSONArray,
+         callbackContext: CallbackContext
+   ): Boolean {
+      if (action == "signIn") {
```
If strategy seems pretty ugly, lets use when statement

Jaquilar replied:
Done

Jaquilar committed:
Merge branch 'developer' into task/integrate-location-module

Jaquilar committed:
Merge branch 'developer' into task/integrate-location-module

Jaquilar committed:
[Improvement] SMP-953: Implementar Delegation Pattern al plugin de Cordova

-  Implemented ModuleDelegate interface

- Updated package name

- Delegate the execute method on each module
